src/bundle.rs

```diff
@@ -325,3 +335,101 @@ pub struct BundleCommitment;
+
+    use crate::{
+        note::{
+            commitment::ExtractedNoteCommitment, nullifier::testing::arb_nullifier,
+            testing::arb_note, TransmittedNoteCiphertext,
+        },
+        primitives::redpallas::testing::arb_spendauth_verification_key,
+        value::{
+            testing::arb_note_value, NoteValue, ValueCommitTrapdoor, ValueCommitment, ValueSum,
+        },
+        Anchor,
+    };
+
+    use super::{Action, Authorization, Bundle, Flags};
+
+    /// Marker for an unauthorized bundle with no proofs or signatures.
+    #[derive(Debug)]
+    pub struct Unauthorized;
+
+    impl Authorization for Unauthorized {
+        type SpendAuth = ();
+    }
+
+    prop_compose! {
+        /// Generate an action without authorization data.
+        pub fn arb_unauthorized_action(value: NoteValue)(
+            nf in arb_nullifier(),
+            rk in arb_spendauth_verification_key(),
+            note in arb_note(value),
+        ) -> Action<()> {
+            let cmx = ExtractedNoteCommitment::from(note.commitment());
+            let cv_net = ValueCommitment::derive(
+                (note.value() - NoteValue::zero()).unwrap(),
+                ValueCommitTrapdoor::zero()
+            );
+            // FIXME: make a real one from the note.
+            let encrypted_note = TransmittedNoteCiphertext {
+                epk_bytes: [0u8; 32],
+                enc_ciphertext: [0u8; 580],
+                out_ciphertext: [0u8; 80]
+            };
+            Action {
+                nf,
+                rk,
+                cmx,
+                encrypted_note,
+                cv_net,
+                authorization: ()
+            }
+        }
+    }
+
+    prop_compose! {
+        /// Create an arbitrary set of flags.
+        pub fn arb_flags()(spends_enabled in prop::bool::ANY, outputs_enabled in prop::bool::ANY) -> Flags {
+            Flags::from_parts(spends_enabled, outputs_enabled)
+        }
+    }
+
+    prop_compose! {
+        /// Generate an arbitrary unauthorized bundle. This bundle does not
+        /// necessarily respect consensus rules; for that use
+        /// [`crate::builder::testing::arb_bundle`]
+        pub fn arb_unauthorized_bundle()(
+            acts in vec(
+                arb_note_value().prop_flat_map(|v|
+                    arb_unauthorized_action(v).prop_map(move |a| (v, a))
+                ),
+                1..10
+            ),
+            flags in arb_flags(),
+            anchor in prop::array::uniform32(prop::num::u8::ANY).prop_map(Anchor)
+        ) -> Bundle<Unauthorized, ValueSum> {
+            let (values, actions): (Vec<NoteValue>, Vec<Action<()>>) = acts.into_iter().unzip();
+
+            Bundle::from_parts(
+                NonEmpty::from_vec(actions).unwrap(),
+                flags,
+                values.into_iter().fold(
+                    ValueSum::zero(),
+                    |acc, cv| (acc + (cv - NoteValue::zero()).unwrap()).unwrap()
```

Comment:
What is `- NoteValue::zero()` doing?

Reply:
`cv` is misnamed here; these are `NoteValue`s, not `ValueCommitment`s. The subtraction is necessary to get a `ValueSum`.